packages/noports_core/lib/src/sshnp/impl/sshnp_dart_pure_impl.dart

```diff
@@ -0,0 +1,101 @@
+import 'dart:async';
+
+import 'package:at_client/at_client.dart';
+import 'package:dartssh2/dartssh2.dart';
+import 'package:noports_core/sshnp_foundation.dart';
+
+class SshnpDartPureImpl extends SshnpCore
+    with SshnpDartSshKeyHandler, SshnpDartInitialTunnelHandler {
```

Comment:
Lots of code duplication between this and sshnp_dart_local_impl.dart - can it be refactored?

Reply:
So I thought the same thing at first, but I decided not to (at least for now). In the current abstraction we defer to the run() function to be an explicit orchestration definition. As I went to try and reduce repetitiveness here, I realized that it might be better to be explicit. For one, this way it's a little more clear what is going on under the hood. and changing just one of these steps is a very likely possibility as more use cases come along.

Reply:
👍